lib/alephant/broker/request/batch.rb

```diff
@@ -10,17 +10,17 @@ class Batch
         attr_reader :batch_id, :components, :load_strategy
 
         def initialize(component_factory, env)
-          logger.info "Request::Batch#initialize: id: #{env.data['batch_id']}"
+          logger.info "Request::Batch#initialize: id: #{env.data['batch_id']}" if env.data
```

Comment:
we still need to consider this given the polling broker performance requirements.